tool/formula/src/test/java/org/openscience/cdk/tools/manipulator/MolecularFormulaManipulatorTest.java

```diff
@@ -1483,4 +1483,19 @@ public void roundtrip(String mfStr, String expected) {
         roundtrip("[2H]2+2", "[[2]H2]2+");
         roundtrip("H+", "[H]+");
     }
+
+    @Test public void atomCountIsSafeForOverflow() {
+        // max integer
+        assertFormulaContainsExactlyAtoms(getMolecularFormula("H2147483647", builder), Integer.MAX_VALUE);
```

Comment:
I don't understand this test. Should there not 2147483647 atoms here?

Reply:
2147483647 is Interger.MAX